Source/SwiftLintBuiltInRules/Rules/Lint/UnneededOverrideRule.swift

```diff
@@ -16,103 +16,136 @@ struct UnneededOverrideRule: Rule {
     )
 }
 
-private func simplify(_ node: some SyntaxProtocol) -> (any ExprSyntaxProtocol)? {
-    if let expr = node.as(AwaitExprSyntax.self) {
-        return expr.expression
-    } else if let expr = node.as(TryExprSyntax.self) {
-        // Assume using try! / try? changes behavior.
-        if expr.questionOrExclamationMark != nil {
-            return nil
+private extension UnneededOverrideRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visitPost(_ node: FunctionDeclSyntax) {
+            if node.isUnneededOverride {
+                self.violations.append(node.positionAfterSkippingLeadingTrivia)
+            }
         }
 
-        return expr.expression
-    } else if let expr = node.as(FunctionCallExprSyntax.self) {
-        return expr
-    } else if let stmt = node.as(ReturnStmtSyntax.self) {
-        return stmt.expression
+        override func visitPost(_ node: InitializerDeclSyntax) {
+            if configuration.affectInits && node.isUnneededOverride {
+                self.violations.append(node.positionAfterSkippingLeadingTrivia)
+            }
+        }
     }
 
-    return nil
-}
+    final class Rewriter: ViolationsSyntaxRewriter {
+        override func visit(_ node: FunctionDeclSyntax) -> DeclSyntax {
+            visitOverridable(node) ?? super.visit(node)
+        }
 
-private func extractFunctionCallSyntax(_ node: some SyntaxProtocol) -> FunctionCallExprSyntax? {
-    // Extract the function call from other expressions like try / await / return.
-    // If this returns a non-super calling function, it will get filtered out later.
-    var syntax = simplify(node)
-    while let nestedSyntax = syntax {
-        if nestedSyntax.as(FunctionCallExprSyntax.self) != nil {
-            break
+        override func visit(_ node: InitializerDeclSyntax) -> DeclSyntax {
+            visitOverridable(node) ?? super.visit(node)
         }
 
-        syntax = simplify(nestedSyntax)
+        private func visitOverridable(_ node: some OverridableDeclSyntax) -> DeclSyntax? {
+            guard node.isUnneededOverride else { return nil }
+            correctionPositions.append(node.positionAfterSkippingLeadingTrivia)
+            let expr: DeclSyntax = ""
+            return expr
+                .with(\.leadingTrivia, node.leadingTrivia)
+                .with(\.trailingTrivia, node.trailingTrivia)
+        }
     }
+}
 
-    return syntax?.as(FunctionCallExprSyntax.self)
+private protocol OverridableDeclSyntax: DeclSyntaxProtocol {
```

Comment:
```suggestion
private protocol OverridableDeclSyntax: WithAttributesSyntax, WithModifiersSyntax {
```

saves you two declarations in the protocol. 😉